diff --git a/arch/arm64/boot/dts/renesas/rz-smarc-common.dtsi b/arch/arm64/boot/dts/renesas/rz-smarc-common.dtsi
index 528c64c..39239e7 100644
--- a/arch/arm64/boot/dts/renesas/rz-smarc-common.dtsi
+++ b/arch/arm64/boot/dts/renesas/rz-smarc-common.dtsi
@@ -240,6 +240,17 @@
 	bus-width = <4>;
 	status = "okay";
 	no-1-8-v;
+
+	non-removable;
+	clock-frequency=<40000000>;
+	max-frequency=<40000000>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+	brcmf: wifi@1 {
+		reg = <1>;
+		compatible = "brcm,bcm4329-fmac";
+		laird,regdomain = "US";
+	};
 };
 
 #if (!MTU3_PWM_SUPPORT)
diff --git a/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi b/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
index af34d5e524d5..fbb4f717f748 100644
--- a/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
@@ -278,18 +278,10 @@ qspi0-ctrl {
 		};
 	};
 
-	/*
-	 * SD0 device selection is XOR between GPIO_SD0_DEV_SEL and SW1[2]
-	 * The below switch logic can be used to select the device between
-	 * eMMC and microSD, after setting GPIO_SD0_DEV_SEL to high in DT.
-	 * SW1[2] should be at position 2/OFF to enable 64 GB eMMC
-	 * SW1[2] should be at position 3/ON to enable uSD card CN3
-	 */
-	sd0-dev-sel-hog {
-		gpio-hog;
+	wl_reg_on {
 		gpios = <RZG2L_GPIO(41, 1) GPIO_ACTIVE_HIGH>;
 		output-high;
-		line-name = "sd0_dev_sel";
+		line-name = "wl_reg_on";
 	};
 
 	sdhi0_emmc_pins: sd0emmc {
