diff --git a/r9a07g044.dtsi.orig b/r9a07g044.dtsi
index 16ff035..727815c 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g044.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a07g044.dtsi
@@ -951,7 +951,7 @@
 				     <GIC_SPI 85 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 86 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "mux", "fil", "arp_ns";
-			phy-mode = "rgmii";
+			phy-mode = "rgmii-id";
 			clocks = <&cpg CPG_MOD R9A07G044_ETH0_CLK_AXI>,
 				 <&cpg CPG_MOD R9A07G044_ETH0_CLK_CHI>,
 				 <&cpg CPG_CORE R9A07G044_CLK_HP>;
@@ -971,7 +971,7 @@
 				     <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 89 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "mux", "fil", "arp_ns";
-			phy-mode = "rgmii";
+			phy-mode = "rgmii-id";
 			clocks = <&cpg CPG_MOD R9A07G044_ETH1_CLK_AXI>,
 				 <&cpg CPG_MOD R9A07G044_ETH1_CLK_CHI>,
 				 <&cpg CPG_CORE R9A07G044_CLK_HP>;

