diff --git a/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi b/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
index c5973d8..de91e8c 100644
--- a/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
@@ -126,8 +126,7 @@
 	status = "okay";
 
 	phy0: ethernet-phy@7 {
-		compatible = "ethernet-phy-id0022.1640",
-			     "ethernet-phy-ieee802.3-c22";
+		compatible = "ethernet-phy-ieee802.3-c22";
 		reg = <7>;
 		rxc-skew-psec = <2400>;
 		txc-skew-psec = <2400>;
@@ -142,8 +141,10 @@
 		txd2-skew-psec = <0>;
 		txd3-skew-psec = <0>;
 
-		interrupt-parent = <&pinctrl>;
-		interrupts = <RZG2L_GPIO(1, 0) IRQ_TYPE_LEVEL_LOW>;
+        vsc8531,vddmac = <1800>;
+        vsc8531,edge-slowdown = <0>;
+        vsc8531,led-0-mode = <VSC8531_LINK_1000_ACTIVITY>;
+        vsc8531,led-1-mode = <VSC8531_LINK_100_ACTIVITY>;
 	};
 };
 
@@ -155,8 +156,7 @@
 	status = "okay";
 
 	phy1: ethernet-phy@7 {
-		compatible = "ethernet-phy-id0022.1640",
-			     "ethernet-phy-ieee802.3-c22";
+		compatible = "ethernet-phy-ieee802.3-c22";
 		reg = <7>;
 		rxc-skew-psec = <2400>;
 		txc-skew-psec = <2400>;
@@ -171,8 +171,10 @@
 		txd2-skew-psec = <0>;
 		txd3-skew-psec = <0>;
 
-		interrupt-parent = <&pinctrl>;
-		interrupts = <RZG2L_GPIO(1, 1) IRQ_TYPE_LEVEL_LOW>;
+		vsc8531,vddmac = <1800>;
+        vsc8531,edge-slowdown = <0>;
+        vsc8531,led-0-mode = <VSC8531_LINK_1000_ACTIVITY>;
+        vsc8531,led-1-mode = <VSC8531_LINK_100_ACTIVITY>;
 	};
 };
 
