diff --git a/r9a07g044l2-smarc.dts.orig b/r9a07g044l2-smarc.dts
index bc2af6c..82ba67e 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc.dts
+++ b/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc.dts
@@ -11,8 +11,32 @@
 #include "rzg2l-smarc-pinfunction.dtsi"
 #include "rz-smarc-common.dtsi"
 #include "rzg2l-smarc.dtsi"
+#include <dt-bindings/net/mscc-phy-vsc8531.h>
 
 / {
 	model = "Renesas SMARC EVK based on r9a07g044l2";
 	compatible = "renesas,smarc-evk", "renesas,r9a07g044l2", "renesas,r9a07g044";
 };
+
+&eth0 {
+        phy0: ethernet-phy@0 {
+                compatible = "ethernet-phy-ieee802.3-c45";
+                reg = <0>;
+                vsc8531,vddmac          = <1800>;
+                vsc8531,edge-slowdown   = <0>;
+                vsc8531,led-0-mode      = <VSC8531_LINK_1000_ACTIVITY>;
+                vsc8531,led-1-mode      = <VSC8531_LINK_100_ACTIVITY>;
+       };
+};
+
+&eth1 {
+        phy1: ethernet-phy@0 {
+                compatible = "ethernet-phy-ieee802.3-c45";
+                reg = <0>;
+                vsc8531,vddmac          = <1800>;
+                vsc8531,edge-slowdown   = <0>;
+                vsc8531,led-0-mode      = <VSC8531_LINK_1000_ACTIVITY>;
+                vsc8531,led-1-mode      = <VSC8531_LINK_100_ACTIVITY>;
+        };
+};
+
