diff --git a/r9a07g044l2-smarc.dts.orig b/r9a07g044l2-smarc.dts
index 651aa6b..f165a7a 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc.dts
+++ b/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc.dts
@@ -17,3 +17,33 @@
 	compatible = "renesas,smarc-evk", "renesas,r9a07g044l2", "renesas,r9a07g044";
 };
 
+&i2c3 {
+        gpio-mistysom: i2c_gpio_expander@20{
+                #gpio-cells = <2>;
+                #interrupt-cells = <2>;
+                compatible = "semtech,sx1502q";
+                reg = <0x20>;
+                gpio-controller;
+                pinctrl-names = "default";
+                pinctrl-0 = <&gpio1_cfg_pins>;
+                gpio1_cfg_pins: gpio1-cfg {
+                        pins = "gpio1";
+                        bias-pull-up;
+                };
+        };
+        gpio-mistycarrier: i2c_gpio_expander@21{
+                #gpio-cells = <2>;
+                #interrupt-cells = <2>;
+                compatible = "semtech,sx1502q";
+                reg = <0x21>;
+                gpio-controller;
+                pinctrl-names = "default";
+                pinctrl-0 = <&gpio2_cfg_pins>;
+                gpio2_cfg_pins: gpio2-cfg {
+                        pins = "gpio2";
+                        bias-pull-up;
+                };
+        };
+};
+
+
