diff --git a/arch/arm64/boot/dts/renesas/rzg2l-smarc-pinfunction.dtsi b/arch/arm64/boot/dts/renesas/rzg2l-smarc-pinfunction.dtsi
index 1c3376b89368..816eacdd8f32 100644
--- a/arch/arm64/boot/dts/renesas/rzg2l-smarc-pinfunction.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2l-smarc-pinfunction.dtsi
@@ -87,12 +87,12 @@ sd1_ctrl {
 	sdhi1_pins_uhs: sd1_uhs {
 		sd1_data_uhs {
 			pins = "SD1_DATA0", "SD1_DATA1", "SD1_DATA2", "SD1_DATA3";
-			power-source = <1800>;
+			power-source = <3300>;
 		};
 
 		sd1_ctrl_uhs {
 			pins = "SD1_CLK", "SD1_CMD";
-			power-source = <1800>;
+			power-source = <3300>;
 		};
 	};
 
diff --git a/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi b/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
index 7a138ba74265..628c3c78d9a9 100644
--- a/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
@@ -299,12 +299,12 @@ sd0_ctrl {
 	sdhi0_pins_uhs: sd0_uhs {
 		sd0_data_uhs {
 			pins = "SD0_DATA0", "SD0_DATA1", "SD0_DATA2", "SD0_DATA3";
-			power-source = <1800>;
+			power-source = <3300>;
 		};
 
 		sd0_ctrl_uhs {
 			pins = "SD0_CLK", "SD0_CMD";
-			power-source = <1800>;
+			power-source = <3300>;
 		};
 	};
 };
