diff --git a/r9a07g044l2-smarc.dts.orig b/r9a07g044l2-smarc.dts
index bc2af6c..ef9665a 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc.dts
+++ b/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc.dts
@@ -16,3 +16,144 @@
 	model = "Renesas SMARC EVK based on r9a07g044l2";
 	compatible = "renesas,smarc-evk", "renesas,r9a07g044l2", "renesas,r9a07g044";
 };
+
+&pinctrl {
+    /delete-node/ can0-stb;
+    /delete-node/ can1-stb;
+    /delete-node/ gpio-sd0-pwr-en-hog;
+    /delete-node/ sd0-dev-sel-hog;
+    /delete-node/ sd1-pwr-en-hog;
+    /delete-node/ i2c2;
+    /delete-node/ scif2;
+    /delete-node/ scif3;
+    /delete-node/ adv7535;
+    /delete-node/ poegd;
+    /delete-node/ gpt@10048400;
+    /delete-node/ gpt4_pins;
+    /delete-node/ gpt4;
+    /delete-node/ sdhi0;
+    /delete-node/ ethernet@11c20000;
+    /delete-node/ eth0_pins;
+    /delete-node/ eth0;
+
+    gpio-hdr-pin11 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(42, 3) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin11";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+    gpio-hdr-pin13 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(42, 4) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin13";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+    gpio-hdr-pin19 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(48, 0) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin11";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+    gpio-hdr-pin21 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(48, 1) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin21";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+    gpio-hdr-pin23 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(48, 3) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin23";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+    gpio-hdr-pin25 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(48, 4) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin25";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+    gpio-hdr-pin33 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(0, 0) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin33";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+    gpio-hdr-pin34 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(43, 0) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin34";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+    gpio-hdr-pin35 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(0, 1) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin35";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+    gpio-hdr-pin36 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(43, 1) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin36";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+    gpio-hdr-pin37 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(1, 0) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin37";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+    gpio-hdr-pin38 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(43, 2) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin38";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+    gpio-hdr-pin40 {
+        gpio-hog;
+        gpios = <RZG2L_GPIO(43, 3) GPIO_ACTIVE_HIGH>;
+        output-high;
+        line-name = "gpio_hdr_pin40";
+        power-source = <3300>;
+        status = "okay";
+    };
+
+};
+
+
+
